File: 4_py/primes.py
import math

def calcFirstNPrimes(n):
    primes = [2]
    i = 2
    while len(primes) < n:
        prime = True
        iRoot = math.sqrt(i) + 1
        for j in range(2, math.ceil(iRoot)):
            if (i % j == 0):
                prime = False
                break;
        if prime:
            primes.append(i)
        i += 1
    return primes

File: 3_js/primes.js
/**
 * Calculate the first n primes
 * from https://stackoverflow.com/a/35112760
 * @param {number} n Number of primes to return
 * @returns {number[]} List of first n primes
 */
function calcFirstNPrimes(n) {
    const primes = [2];
    for (let i=2; primes.length<n; i++){
        let prime = true;
        let iRoot = Math.sqrt(i) + 1;
        for (let j=2; j<iRoot; j++) {
            if (i % j === 0) {
                prime=false;
                break;
            }
        }
        if (prime) {
            primes.push(i);
        }
    }
    return primes;
}

export {calcFirstNPrimes};